### DISCUSSION TITLE
Update to ubutu 20.04 and NVIDIA toolkit 11.2

### DISCUSSION THREAD

**Discussion by EasyIP2023** [OE4T/meta-tegra#579] [Ideas]
Want to update the [recipes-devtools/cuda](https://github.com/OE4T/meta-tegra/tree/master/recipes-devtools/cuda) folder to work with ubuntu 20.04 and Nvida toolkit 11.2. There's a lot of sha256 and md5 hashes to update, just want to be clear on which files, most likely all, that need to be updated to make this happen. Still new to yocto. Actually needed to switch to ubuntu18.04 to get a working build. My updates to get it working with Nvidia toolkit 10.2 and ubuntu 18.04 are bellow.

```
mkdir -p ~/Downloads/nvidia/sdkm_downloads
wget https://developer.download.nvidia.com/compute/cuda/10.2/Prod/local_installers/cuda-repo-ubuntu1804-10-2-local-10.2.89-440.33.01_1.0-1_amd64.deb -O ~/Downloads/nvidia/sdkm_downloads/cuda-repo-ubuntu1804-10-2-local-10.2.89-440.33.01_1.0-1_amd64.deb
```

Update ```tegra-demo-distro/layers/meta-tegra/recipes-devtools/cuda/cuda-binaries-common-defs.inc``` to
```
CUDA_REPO_EXTRA_x86-64 = "-10.2.89-440.33.01"
```

Update ```tegra-demo-distro/layers/meta-tegra/recipes-devtools/cuda/cuda-binaries-ubuntu1804_10.2.89-1.bbc``` to
```
SRC_URI = "${NVIDIA_DEVNET_MIRROR}/cuda-repo-ubuntu1804-10-2-local-10.2.89-440.33.01_1.0-1_amd64.deb"
SRC_URI[sha256sum] = "a9a5ab0324291b25170245ad39817684487f9bceda1848f05be1b53acd55fafc"
```

**Comment by madisongh** [comment]
If you're cross-building for the Jetson platforms, you really should be using the tools from JetPack, and not downloading from the CUDA site.  

As far as the OE/Yocto build environment goes, it doesn't matter what version of Ubuntu (or whatever host OS) you have.  The JetPack-supplied host-side tools work fine on later versions of Ubuntu (I'm using 20.10).

I use an lxc/lxd container running 18.04 to run the NVIDIA SDK Manager for downloads. I don't have links handy, but there are articles around the Internet on how to get GUI apps to work in such containers. A VM running 18.04 should work fine, too.
